### DISCUSSION TITLE
Expected object, received value and deserialization put the value into object child

### DISCUSSION THREAD

**Discussion by Kellasone** [FasterXML/jackson-databind#4401] [Q&A]
Hello. I want to ask a question about a situation i run into and i cant find something already explained about this.
Lets consider the following object: 

`public Parent parent(){
@NonNull
public Boolean child;
}`

In my request class i have this object with @Valid if that is of any importance, as follows: 
`@Valid
public Parent parent`

The scenario is the following: 
In the request, If i send 
`"parent":true`
its mapped like i would sent 
`"parent":{ "child": true }`

LE: 
Target POJO: 
```
import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.constraints.NotNull;
import lombok.*;
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonInclude(NON_NULL)
public class Parent {
  @NotNull
  private Boolean child;
}
```

Test class created to explain the situation: 
```
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
class ParentTest {
  private ObjectMapper objectMapper;
  @Test
  public void test() throws JsonProcessingException {
    objectMapper = new ObjectMapper();
    var obj = objectMapper.readValue("{\"child\":true}", Parent.class);
    System.out.println(obj);
    var obj2 = objectMapper.readValue("true", Parent.class);
    System.out.println(obj2);
  }
}
```

The problem comes from obj2, when I pass only a boolean value instead of a key child:boolean value, and the output is the same as obj. 
I look for an out of the box solution to stop this and throw an error when this happens.
End of L.E.

My expected behaviour is to fail due to passing a value instead of an object. 
Is there any idea if i can stop this value propagation at deserialization with a solution out of the box? (a feature maybe but i cannot find any to do what i need)
I dont know if this is a databind problem or validation problem, still trying to pinpoint the problem.

More details which might be usefull:
Spring, SpringBoot 3.2.1, JacksonDatabind 2.14.1,

**Comment by yawkat** [comment]
can you make a test case? what you show is not valid java

**Reply by Kellasone** [reply]
Updated the Question with a proper java example.

**Comment by cowtowncoder** [comment]
That "test" class doesn't test anything but prints out stuff. What is the problem?

And no, `@Valid` is not something Jackson understands: it is something a validator (Bean Validation API implementation) would apply on POJOs.

**Reply by Kellasone** [reply]
That class test was not meant to test anything but rather show what i want to say.
Those 2 printed objects are identical even tho the input is different. 
The problem is the second object, obj2. I want to stop that behavior where the input is bad but the mapper does some magic to transform it into a valid object.

**Reply by cowtowncoder** [reply]
It looks like Lombok was used. It would be necessary to know exactly how class looks like after Lombok processing.

I suspect there is a Constructor (generated by Lombok) that is discovered as "delegating" Creator method: takes single-boolean/Boolean argument, which then would match JSON Boolean value.
To prevent that you'd either prevent such constructor from being generated, or mark it with `@JsonIgnore`.
Or, to make sure it is used as "properties-based" and not "delegating", annotate with

    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)

FWTW, cases like this can be one downside to frameworks like Lombok as they add further complexity and opaqueness to scenarios.
